### PR TITLE
Initial sample

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* This PR adds an Electron desktop app that is calling MS Graph API with MSAL Node using auth code with PKCE.
* This sample is meant to be used as a **quickstart** sample.

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ x ] New sample
```

## How to Test
*  Get the code

```
git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git
cd ms-identity-javascript-nodejs-desktop
git checkout sample
npm install
```

cc @hectormmg

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,57 +1,77 @@
...
+* [Visual Studio Code](https://code.visualstudio.com/download) is recommended for running and editing this sample.
 
-### Quickstart
-(Add steps to get up and running quickly)
+### Setup
 
-1. git clone [repository clone url]
-2. cd [respository name]
-3. ...
+1. [Register a new application](https://docs.microsoft.com/azure/active-directory/develop/scenario-desktop-app-registration) in the [Azure Portal](https://portal.azure.com).
```
Comment (idg-sam) [inline README.md]:
```suggestion
1. Register a [desktop app that calls web APIs](https://docs.microsoft.com/azure/active-directory/develop/scenario-desktop-app-registration) in the [Azure Portal](https://portal.azure.com).
```

Comment (idg-sam) [reply README.md]: minor change to link

File: README.md [hunk 2]
```diff
@@ -1,57 +1,77 @@
...
+1. Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
```
Comment (idg-sam) [inline README.md:54]:
```suggestion
    - Navigate to the Authentication blade in the app registration.
    - Click add a platform and choose **Mobile and Desktop application.**
    - For the **redirect URI**, choose custom redirect URI. Enter `msal://redirect` into the input box.
1. Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
```

**Review by Shama-K — COMMENTED** [review]
Sign-in with a user account and sign-out. Click Sign in button, it will sign-in automatically with the same user.

File: README.md [hunk 3]
```diff
@@ -1,57 +1,80 @@
...
-### Quickstart
-(Add steps to get up and running quickly)
+### Setup
 
-1. git clone [repository clone url]
-2. cd [respository name]
-3. ...
+1. Register a [desktop app that calls web APIs](https://docs.microsoft.com/azure/active-directory/develop/scenario-desktop-app-registration) in the [Azure Portal](https://portal.azure.com).
+    - Navigate to the Authentication blade in the app registration.
+    - Click add a platform and choose **Mobile and Desktop application.**
```
Comment (idg-sam) [inline README.md]:
```suggestion
    - Be sure to **remove** any platforms registered here (e.g. web application, SPA, etc)
    - Click add a platform and choose **Mobile and Desktop application.**
```

File: README.md [hunk 4]
```diff
@@ -1,57 +1,80 @@
...
+- azure-active-directory
+description: "Demonstrates how to use MSAL Node to sign-in users and acquire access tokens for a protected resource such as Microsoft Graph in an Electron desktop application using the OAuth 2.0 authorization code flow with PKCE."
+urlFragment: "ms-identity-javascript-nodejs-desktop"
+---
 
-(short, 1-3 sentenced, description of the project)
+# An Electron desktop application secured by MSAL Node (Preview) on Microsoft identity platform
 
-## Features
+This sample demonstrates how to use [MSAL Node (Preview)](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-node) to sign-in a user and acquire an access token for a protected resource such as Microsoft Graph in an Electron desktop application using the [authorization code grant with PKCE](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-auth-code-flow).
```
Comment (kalyankrishna1) [inline README.md]: >with PKCE [](start = 315, length = 9)

with PKCE flow

File: README.md [hunk 5]
```diff
@@ -1,57 +1,80 @@
...
+## Features
 
-## Getting Started
+This sample demonstrates the following **MSAL Node** concepts:
 
-### Prerequisites
+* Configuration
+* Login and logout
+* Acquiring an access token
+* Calling a web API
```
Comment (kalyankrishna1) [inline README.md:27]: are these four documented somewhere? or are they links to sections in README?

File: README.md [hunk 6]
```diff
@@ -1,57 +1,80 @@
...
+    * For the **redirect URI**, enter `msal://redirect` into the input box.
+1. Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
+1. Open the [.env](.env) file and provide the required configuration values.
+    1. Replace the string `Enter_the_Application_Id_Here` with your app/client ID on Azure AD portal.
+    1. Replace the string `Enter_the_Tenant_Id_Here` with your tenant ID on Azure AD portal.
+    1. Replace the string `Enter_the_Client_Secret_Here` with the client secret you created on Azure AD portal.
+    1. Replace the string `Enter_the_Cloud_Instance_Id_Here` with `https://login.microsoftonline.com/` (see **note** below).
+    1. Replace the string `Enter_the_Graph_Endpoint_Here`. with `https://graph.microsoft.com/` (see **note** below).
 
+> :information_source: *note*: This is for multi-tenant applications located on the global Azure cloud. For more information, see: [Use MSAL in a national cloud environment](https://docs.microsoft.com/azure/active-directory/develop/quickstart-v2-javascript-auth-code)
```
Comment (kalyankrishna1) [inline README.md]: >global [](start = 84, length = 6)

global->Global

Comment (kalyankrishna1) [inline README.md]: >global Azure cloud [](start = 84, length = 18)

link to [https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud)

File: README.md [hunk 7]
```diff
@@ -1,57 +1,80 @@
...
-3. ...
+1. Register a [desktop app that calls web APIs](https://docs.microsoft.com/azure/active-directory/develop/scenario-desktop-app-registration) in the [Azure Portal](https://portal.azure.com).
+    * Navigate to the **Authentication** blade in the app registration.
+    * Click add a platform and choose **Mobile and Desktop application**
+    * For the **redirect URI**, enter `msal://redirect` into the input box.
+1. Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
+1. Open the [.env](.env) file and provide the required configuration values.
+    1. Replace the string `Enter_the_Application_Id_Here` with your app/client ID on Azure AD portal.
+    1. Replace the string `Enter_the_Tenant_Id_Here` with your tenant ID on Azure AD portal.
+    1. Replace the string `Enter_the_Client_Secret_Here` with the client secret you created on Azure AD portal.
```
Comment (kalyankrishna1) [inline README.md]: >Replace the string `Enter_the_Client_Secret_Here` with the client secret you created on Azure AD portal. [](start = 7, length = 104)

it does not exists

**Review by kalyankrishna1 — CHANGES_REQUESTED** [review]
:clock1:

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

**Review by samuelkubai — APPROVED** [review]
LGTM

**Review by hectormmg — APPROVED** [review]
Looks great!

**Pull request merged**